File: src/main/java/net/slipp/domain/fb/FacebookComment.java
package net.slipp.domain.fb;

import java.util.Date;

import net.slipp.domain.tag.Tag;
import net.slipp.support.web.tags.SlippFunctions;

import com.restfb.types.CategorizedFacebookType;
import com.restfb.types.Comment;

public class FacebookComment implements Comparable<FacebookComment> {
    private String id;
    private String userId;
    private String name;
    private Date createdTime;
    private String message;
    private String groupId;
    private String groupName;

    private FacebookComment(String id, String userId, String name, Date createdTime, String message, String groupId, String groupName) {
        this.id = id;
        this.userId = userId;
        this.name = name;
        this.createdTime = createdTime;
        this.message = message;
        this.groupId = groupId;
        this.groupName = groupName;
    }

    public static FacebookComment create(Tag tag, Comment comment) {
        CategorizedFacebookType user = comment.getFrom();
        
        if (tag == null) {
        	return new FacebookComment(comment.getId(), user.getId(), user.getName(), comment.getCreatedTime(), comment.getMessage(), null, null);
        } 
        
        return new FacebookComment(comment.getId(), user.getId(), user.getName(), comment.getCreatedTime(), comment.getMessage(), tag.getGroupId(), tag.getName());
    }

    public String getId() {
        return id;
    }

    public Date getCreatedTime() {
        return createdTime;
    }

    public String getMessage() {
        return SlippFunctions.hbr(message);
    }

    public String getUserId() {
        return userId;
    }

    public String getName() {
        return name;
    }
    
    public String getGroupId() {
		return groupId;
	}
    
    public String getGroupName() {
		return groupName;
	}
    
    @Override
    public int compareTo(FacebookComment target) {
        return (int)(createdTime.getTime() - target.createdTime.getTime());
    }

    @Override
    public String toString() {
        return "FacebookComment [id=" + id + ", userId=" + userId + ", name=" + name + ", createdTime=" + createdTime
                + ", message=" + message + "]";
    }
}

File: src/main/scala/net/slipp/domain/fb/FacebookComment.scala
package net.slipp.domain.fb

import java.util.Date

import com.restfb.types.{CategorizedFacebookType, Comment}
import net.slipp.domain.tag.Tag
import net.slipp.support.web.tags.SlippFunctions.{br, links}

import scala.beans.BeanProperty

object FacebookComment {
  def create(tag: Tag, comment: Comment): FacebookComment = {
    val user: CategorizedFacebookType = comment.getFrom
    if (tag == null) {
      return new FacebookComment(comment.getId, user.getId, user.getName, comment.getCreatedTime, comment.getMessage, null, null)
    }
    return new FacebookComment(comment.getId, user.getId, user.getName, comment.getCreatedTime, comment.getMessage, tag.getGroupId, tag.getName)
  }
}

case class FacebookComment(id: String, userId: String, name: String, createdTime: Date, message: String, groupId: String, groupName: String)
  extends Comparable[FacebookComment] {
  def getMessage = {
    br(links(message))
  }

  def compareTo(target: FacebookComment) = {
    (createdTime.getTime - target.createdTime.getTime).toInt
  }

  override def toString: String = {
    return "FacebookComment [id=" + id + ", userId=" + userId + ", name=" + name + ", createdTime=" + createdTime + ", message=" + message + "]"
  }
}
